jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -338,14 +334,21 @@ def _calculate_statistics(self, msg={}, store_results=False):
 
         spectrum, continuum, spec_subtracted = self._get_continuum(self.dataset,
                                                                    self.spectral_subset,
-                                                                   update_marks=True)
+                                                                   update_marks=update_marks)
+
         if spectrum is None:
             self.update_results(None)
             return
 
         if not store_results:
             return
 
+        # if there are no results, statistics do not need to be re-calculated
+        # when is_active is toggled on
+        if hasattr(msg, 'name'):
+            if msg.name == 'is_active' and not self.results_available:
+                return
```

Comment:
(feel free to ignore)

```suggestion
        if getattr(msg, 'name', '') == 'is_active' and not self.results_available:
            return
```